agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -244,6 +245,11 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
             (builder, otelConfig) ->
                 configureMetrics(metricFilters, builder, telemetryClient, configuration));
 
+    AiContextCustomizerHolder.setInstance(
+        new AiContextCustomizer<>(
+            configuration.preview.connectionStringOverrides,
```

Comment:
I don't know if `configuration.preview` could be null

Reply:
it cannot (well, it can, with json configuration below, but this is a problem in lots of places, may want to think about separately)

```
{
  "preview": null
}
```